opencti-platform/opencti-graphql/src/modules/workspace/investigation-domain.ts

```diff
@@ -0,0 +1,109 @@
+import { v4 as uuidv4 } from 'uuid';
+import type { AuthContext, AuthUser } from '../../types/user';
+import type { BasicStoreEntityWorkspace } from './workspace-types';
+import { FunctionalError } from '../../config/errors';
+import { stixLoadByIds } from '../../database/middleware';
+import { convertTypeToStixType } from '../../database/stix-converter';
+import { ENTITY_TYPE_CONTAINER_REPORT } from '../../schema/stixDomainObject';
+import { STIX_SPEC_VERSION } from '../../database/stix';
+import { generateStandardId } from '../../schema/identifier';
+import type { StixId, StixObject } from '../../types/stix-common';
+import { elList } from '../../database/middleware-loader';
+import { READ_INDEX_INTERNAL_OBJECTS } from '../../database/utils';
+import { addWorkspace } from './workspace-domain';
+import type { BasicStoreEntity } from '../../types/store';
+
+const EXPORTABLE_INVESTIGATED_TYPES: string[] = ['Stix-Core-Object', 'Stix-Core-Relationship', 'Stix-Sighting-Relationship'];
+
+type StixReportForExport = {
+  id: StixId;
+  name: string;
+  object_refs: StixId[];
+  published: Date;
+  spec_version: string;
+  type: string
+};
+
+const buildStixReportForExport = (workspace: BasicStoreEntityWorkspace, stixInvestigatedEntities: StixObject[]): StixReportForExport => {
+  const stixReportBundleWithoutId = {
+    name: workspace.name,
+    object_refs: stixInvestigatedEntities.map((s) => s.id),
+    published: workspace.created_at,
+    spec_version: STIX_SPEC_VERSION,
+    type: convertTypeToStixType(ENTITY_TYPE_CONTAINER_REPORT),
+  };
+  const id = generateStandardId(ENTITY_TYPE_CONTAINER_REPORT, stixReportBundleWithoutId) as StixId;
+
+  return { id, ...stixReportBundleWithoutId };
+};
+
+type StixBundle = {
```

Comment:
Maybe not the best place, if it's the representation of a generic stix bundle

Reply:
It's not, see `objects` field of type `(StixObject | StixReportForExport)[]`. I decided it was a better choice to have this definition locally first, then let duplication appearing and finally tend toward a more general type definition.